core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/iterator/CrossProductIteration.java

```diff
@@ -67,4 +67,12 @@ protected BindingSet getNextElement()
 
 		return null;
 	}
+
+	@Override
+	protected void handleClose()
+		throws QueryEvaluationException
+	{
+		super.handleClose();
```

Comment:
That's fine but if we do that we should do it everywhere. Almost all implementations of handleClose() in iterations do the same as I did (some first close the wrapped iterations and then call super.handleClose()). Shouldn't we rather have a more generic way to do this consistently?